src/object.d

```diff
@@ -424,6 +433,807 @@ class TypeInfo
+                assert(id.initializer().length == 1);
+            }
+            else static if (is(T == U[], U))
+            {
+                static assert(id.next == __typeid!U);
+                static if (is(U == int))
+                {
+                    auto a = [ 1, 2, 3 ], b = [ 0, 1, 2, 3, 4 ];
+                    assert(!id.equals(&a, &b));
+                    assert(!id2.equals(&a, &b));
+                    b = b[1 .. 4];
+                    assert(id.equals(&a, &b));
+                    assert(id2.equals(&a, &b));
+                }
+                static assert(id.flags == 1);
+                static assert(id2.flags == 1);
+            }
+            else static if (is(T U == enum))
+            {
+                static assert(id.next == __typeid!U);
+                if (id.initializer.ptr)
+                {
+                    T x1;
+                    T x2 = *cast(T*) id.initializer.ptr;
+                    assert(x1 == x2);// || __traits(isFloating, T));
+                }
+            }
+
+            assert(id.rtInfo == RTInfo!T);
+            assert(id2.rtInfo == RTInfo!T);
+
+            static if (is(T == class))
+            {
+                auto cid = __typeid!T;
+                TypeInfo_Class2 cid2 = __typeid!T;
+                assert(cid.base == cid2.base);
+                assert(cid.interfaces == cid2.interfaces);
+            }
+        }
+    }
+    enum E1 { a, b }
+    enum E2 : float { a, b }
+    enum E3 : cdouble { a = 0 + 42i, b }
+    enum E4 { a = 42, b }
+    struct S1 { double a = 0; }
+    struct S2 { int a; S2* next; }
+    struct S3 { int a; S2[] array; }
+    struct S4 { int a; S2 b; }
+
+    class C1 {}
+    extern(C++) class C2 {}
+
+    interface I1 {
+        void fun();
+    }
+    interface I2 {
+        void gun();
+    }
+    class C3 : I1, I2 {
+        void fun() {}
+        void gun() {}
+        double x;
+    }
+
+    test!(
+        int, const(int), shared(int), inout(int), immutable(int),
+        inout(const int), shared(const int), inout(shared int), const(inout(shared int)),
+        int*, const(int*), shared(int*), inout(int*), immutable(int*),
+        inout(const int*), shared(const int*), inout(shared int*), const(inout(shared int*)),
+        cdouble, ifloat,
+        int*, int[], E1, E2, E3, E4,
+        int[42], int[int], __vector(int[4]),
+        int function(double), string delegate(int[]),
+        S1, const S1, shared S1, const inout shared S1, immutable S1,
+        S2, const S2, shared S2, const inout shared S2, immutable S2,
+        C1, C3, I1, I2,
+        );
+
+    static assert(__typeid!S1.flags == 0);
+    static assert(__typeid!S2.flags == 1);
+    static assert(__typeid!S3.flags == 1);
+    static assert(__typeid!S4.flags == 1);
+
+    // Test a static array with types with destructor.
+    static uint x = 0;
+    static struct A { ~this() { ++x; } }
+    A[42] a;
+    __typeid!(A[42]).destroy(&a);
+    assert(x == 42);
+}
```

Comment:
any chance of more tests to eliminate the (non assert) uncovered lines of code?

Reply:
will try if this pr is to move forward